### PR TITLE
[x] a number of improvements

### PR DESCRIPTION
**Pull request by sunshowers**
Started digging into CI and feature unification for our packages -- here are a few improvements I spotted along the way.

### CODE REVIEW

**Comment by sunshowers** [comment]
@vgao1996, @runtian-zhou, @dariorussi: I'm making a few changes to how some of the stuff in language/ is organized. Please have a look, thanks 💕

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M termcolor 1.1.2 (transitive third-party, crates.io)
    * status changed from direct third-party
    * (unchanged features: [none])
  R datatest-stable 0.1.0 (workspace, path 'common/datatest-stable')
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M termcolor 1.1.2 (transitive third-party, crates.io)
    * status changed from direct third-party
    * (unchanged features: [none])
  R datatest-stable 0.1.0 (workspace, path 'common/datatest-stable')
    * (old features: [none])
```

**Comment by sunshowers** [comment]
Had to reorganize some stuff in move-lang as well, heads up @tnowacki.

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M termcolor 1.1.2 (transitive third-party, crates.io)
    * status changed from direct third-party
    * (unchanged features: [none])
  R datatest-stable 0.1.0 (workspace, path 'common/datatest-stable')
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M termcolor 1.1.2 (transitive third-party, crates.io)
    * status changed from direct third-party
    * (unchanged features: [none])
  R datatest-stable 0.1.0 (workspace, path 'common/datatest-stable')
    * (old features: [none])
```

**Comment by rexhoffman** [comment]
Out of curiosity what sort of tests do you expect people to put in the playground command/how long will they live there?

File: devtools/x-lint/src/lib.rs [hunk 1]
```diff
@@ -190,6 +188,7 @@ pub mod prelude {
         file_path::{FilePathContext, FilePathLinter},
         package::{PackageContext, PackageLinter},
         project::{ProjectContext, ProjectLinter},
+        runner::{LintEngine, LintEngineConfig, LintResults},
```
Comment (rexhoffman) [inline devtools/x-lint/src/lib.rs:191]: what's the impact of this line, what was happening when it wasn't here?   Trying to wrap my head around this.

**Comment by tnowacki** [comment]
> Had to reorganize some stuff in move-lang as well, heads up @tnowacki.

Thanks for the heads up! I wouldn't have expected anything there to need to be relocated. What is the need here? (I wouldn't have expected move-lang to be in the TCB)

File: devtools/x-lint/src/lib.rs (hunk 1 above)
Comment (sunshowers) [reply devtools/x-lint/src/lib.rs:191]: Previously we had made runner private and re-exported all its types to the top level. With this change runner is now public and its types are exported in the prelude. It's a minor change but I think it makes it better.

**Review by sunshowers — COMMENTED** [review]
I don't expect the playground to be used for any real code. just local experimentation. (I used it to figure out instances where we were unifying features a bunch for different ways, it was cool.)

**Comment by sunshowers** [comment]
@tnowacki move-lang isn't in the TCB but also is part of our production-quality code. It was exporting test_utils which required the datatest-stable library. Previously we made the mistake of not treating datatest-stable as test-only: when I fixed that I had to update move-lang as well.

**Comment by tnowacki** [comment]
> It was exporting test_utils which required the datatest-stable library. 
Makes sense to me!

For, ir-utils/src/bin/ir-file-translation.rs, I didn't think this was using datatest, why was it moved too? (not that it matters much)

**Comment by sunshowers** [comment]
> > It was exporting test_utils which required the datatest-stable library. 
> Makes sense to me!
> 
> For, ir-utils/src/bin/ir-file-translation.rs, I didn't think this was using datatest, why was it moved too? (not that it matters much)

Yeah it wasn't necessary but I thought it made sense to group the ir utils together. Lmk if you prefer to not move it

**Comment by vgao1996** [comment]
The part about `stdlib` and `move-cli` looks good to me.

**Review by rexhoffman — APPROVED** [review]
Nothing stands out we didn't already discuss in chat.    Which is really beyond the scope of this pr.

**Pull request closed without merging**